pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -551,16 +623,16 @@ func (nc *NodeController) MonitorNodeStatus() error {
 			continue
 		}
 
-		readyCondition := nc.getCondition(node, api.NodeReady)
+		readyCondition := nc.getCondition(&node.Status, api.NodeReady)
```

Comment:
IIUC, in tryUpdateNodeStatus, if you found the node status hasn't been updated for grace peroid and ready condition is nil, you will append a ready condition to Node.Status.  Since everything is pointer, which means readyCondition will never be nil here. 

It worked before because readyCondition is never re-get from node.Status.